### ISSUE TITLE
[FEATURE REQ] Azure.AI.OpenAI `include_usage` in chat completion options

### ISSUE THREAD

**Issue by janaka** [Azure/azure-sdk-for-net#44237]
### Library name

Azure.AI.OpenAI

### Please describe the feature.

Add `include_usage` options for streaming chat completions. Relevant OpenAI docs below.

https://platform.openai.com/docs/api-reference/completions/create

include_usage

Request:
<img width="676" alt="image" src="https://github.com/Azure/azure-sdk-for-net/assets/51779/0ca809c4-c17b-41a6-b171-e54154097bce">

Respose:
<img width="694" alt="image" src="https://github.com/Azure/azure-sdk-for-net/assets/51779/aedad1ed-8455-4a14-b499-6cc05ee106b9">

**Comment by BlackGad** [comment]
https://github.com/Azure/azure-rest-api-specs/issues/25062 related stuff

**Comment by janaka** [comment]
Updated to the new v2 beta2 client. This feature seems to be switched to `true` by default in the OpenAI client but doesn't seem to be working.

https://github.com/openai/openai-dotnet/issues/103

**Comment by Mako217** [comment]
Are there any updates on this?

**Comment by janaka** [comment]
> Are there any updates on this?

@Mako217 on https://github.com/openai/openai-dotnet/issues/103 I was told the backend doesn't have the feature yet. Generally Azure have been terrible at communicating what the state of these features are. There are at least two other features we are waiting for were the backend state is unclear.

**Comment by janaka** [comment]
yes, confirmed for the save model and rest API version against eastus and sweden deployments. streaming + include_usage works for me with Postman.

**Comment by mirkus90** [comment]
@janaka hi! could you please share the body of your REST request?

I've followed the REST Spec API and created this body, but it seems that the streamOptions parameter has not the correct name

{
  "messages": [
        {
          "role": "system",
          "content": "you are a helpful assistant that talks like a pirate"
        },
        {
          "role": "user",
          "content": "can you tell me how to care for a parrot?"
        }
  ],
  "stream": true,
  "streamOptions": {
    "include_usage": true
  },
  "max_tokens": 800,
  "temperature": 0.7
}

**Comment by janaka** [comment]
We are not on v2.1 of the dotnet SDK and this issue still has not been fixed. 

As documented in https://github.com/openai/openai-dotnet/issues/103 this is a bug in the client. The backend is confirmed to support this feature.

Can somebody on the SDK team at least acknowledge this is an issue cc @jpalvarezl  @trrwilson .

**Comment by janaka** [comment]
Pretty sure this is sorted in v2.2.0 for us so closing. Thanks
